observIQ/bindplane-op
# fix: ApolloClient type policies.

### Proposed Change
We were seeing strange behavior where the apollo cache would merge the `Metadata` values of a Configuration and a SourceType with the same `metadata.name` field.  

- Updated the `Metadata` type policy to specify the combination of `name` and `id` to be the keyfields.
- Make sure we always request `name` and `metadata` fields on gql queries.
- Specified a custom merge function for the `ResourceType.spec` fields of SourceType and DestinationType.  This gets rid of this warning on the Configuration details page: 

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/64915094/220780226-cd7a6b1d-6617-4dc4-995c-d2f814b6d275.png">


### Testing

1. Create a configuration named host
2. Add a host source to the configuration
3. Make sure the description field is not suddenly populated with the description of the Host SourceType.

##### Checklist
